src/poly/sampler.rs

```diff
@@ -52,11 +61,14 @@ pub trait PolyTrapdoorSampler {
     type M: PolyMatrix;
     type Trapdoor;
 
+    fn new(params: &<<Self::M as PolyMatrix>::P as Poly>::Params, sigma: f64) -> Self;
+
     fn trapdoor(
         &self,
         params: &<<Self::M as PolyMatrix>::P as Poly>::Params,
         size: usize,
     ) -> (Self::Trapdoor, Self::M);
+
     fn preimage(
         &self,
         params: &<<Self::M as PolyMatrix>::P as Poly>::Params,
```

Comment:
For each sampler trait, could you define `type Params` and add it to the new function?
This abstraction is necessary because we are not sure if any implementation of the samplers has the common new arguments.

Reply:
actually i don't think having seperate type Params is good idea for sampler. 
lots of Params are shared with sampler and matrix and poly, where if trapdoor define it's own type, there's lots of conflict happen on having generic param but defined different on `Poly` trait level and `Sampler` trait level

This is same reason why we not defined `type Params` on `PolyMatrix` trait and use trait from `Poly`